mne/annotations.py

```diff
@@ -58,7 +58,103 @@
 _datetime = datetime
 
 
-def _check_o_d_s_c(onset, duration, description, ch_names):
+class _AnnotationsExtrasDict(UserDict):
+    """A dictionary for storing extra fields of annotations.
+
+    The keys of the dictionary are strings, and the values can be
+    strings, integers, floats, or None.
+    """
+
+    def __setitem__(self, key: str, value: str | int | float | None) -> None:
+        _validate_type(key, str, "key", "string")
```

Comment:
This makes our behavior in this one function different from elsewhere in MNE-Python
```suggestion
        _validate_type(key, str, "key")
```
If we really want to be more verbose, we should update `_validate_type` to deal with `str` in the list of acceptable types to write it as `"string"` rather than `"str"` so that all places in MNE-Python benefit from it (and consistency is maintained).

But I'm not sure we should... following this logic one could argue you should say "integer" instead of "int", "floating point value" instead of "float", etc...

Reply:
I see, thanks for the lecture :D